src/lib/comms/sol-lwm2m-bs-server.c

```diff
@@ -314,23 +315,31 @@ sol_lwm2m_bootstrap_server_new(uint16_t port, const char **known_clients,
         }
     }
 
-    free(sec_modes);
-    free(cipher_suites);
+    sol_vector_init(&server->known_clients, sizeof(struct sol_str_slice));
 
-    server->known_clients = known_clients;
+    for (i = 0; known_clients[i]; i++) {
+        known_client = sol_vector_append(&server->known_clients);
+        SOL_NULL_CHECK_GOTO(known_client, err_security);
+        *known_client = sol_str_slice_from_str(known_clients[i]);
```

Comment:
We still have the problem, the user can free(known_clients[i]) from outside. Maybe you should create a sol_ptr_vector and strdup() the known_clients.